spec/controllers/shoutouts_controller_spec.rb

```diff
@@ -0,0 +1,219 @@
+require 'rails_helper'
+require 'rspec/json_expectations'
+
+RSpec.describe Api::V1::ShoutoutsController, type: :controller do
+  include ApplicationHelper
+  let!(:user) { create :user }
+  let!(:user1) { create :user }
+  let!(:user2) { create :user }
+  let!(:user3) { create :user }
+  let!(:user4) { create :user }
+  let!(:user5) { create :user }
+  let!(:time_period) { create :time_period }
+  describe 'POST #create' do
+    context 'with valid parameters' do
+      it 'creates a new shout-out with recipients of this shoutout' do
+        allow(controller).to receive(:current_user).and_return(user)
+        expect do
+          post :create, params: { shoutout: { user_id: user[:id],
+                                              time_period_id: time_period[:id],
+                                              rich_text: 'value2',
+                                              recipients: [user2.id, user3.id, user4.id] } }
+        end.to change(Shoutout, :count).by(1)
+        expect(ShoutoutRecipient.all.length).to be(3)
+      end
+
+      it 'not allow duplicate records' do
+        allow(controller).to receive(:current_user).and_return(user1)
+        expect do
+          post :create, params: { shoutout: { user_id: user1[:id],
+                                              time_period_id: time_period[:id],
```

Comment:
the same and please check all similar issues

Reply:
Thanks I fixed it